bluetooth: controller: fixed problems in ISO-AL RX and included RX unit tests

niag-Oticon opened the pull request:
[*] Fixed setting of SDU production status when exiting the
    ISOAL_ERR_SPOOL state for framed PDU consumption when receiving the
    start of the next SDU
[*] Changed condition identifying a padding PDU to include padding PDUs
    with errors received after the end fragment is seen and expanded
    unit tests to cover new condition
[*] Changed definition of ISOAL_PDU_STATUS_xxx errors to match value of
    similar errors defined for the SDU status

Ztest based unit test of ISO adaptation layer.
Testing of unframed case, single SDU, recombined from
one and two PDUs.

mtpr-ot reviewed (dismissed):
LGTM

SebastianBoe commented on tests/bluetooth/ctrl_isoal/CMakeLists.txt:
```diff
@@ -0,0 +1,19 @@
+project(bluetooth_ctrl_isoal)
+
+zephyr_library_include_directories(
+	${ZEPHYR_BASE}/subsys/bluetooth
```
Isn't this already in the include path?

at least it looks that way from

zephyr/subsys/bluetooth/CMakeLists.txt

please update other tests as well if they have the same problem

niag-Oticon replied:
The build fails if this is removed. It does not appear to be in the include path.

SebastianBoe commented on tests/bluetooth/ctrl_isoal/CMakeLists.txt:
```diff
@@ -0,0 +1,19 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.13.1)
+include_directories("./src")
```
You don't hvae any header files in src.

Is someone #including main.c ?

niag-Oticon replied:
Removed. Done.

SebastianBoe reviewed (changes requested):
cmake changes requested

cvinayak commented on subsys/bluetooth/controller/ll_sw/isoal.c:
```diff
@@ -451,9 +452,17 @@ static isoal_status_t isoal_rx_unframed_consume(struct isoal_sink *sink,
+	length = pdu_err ? 0 : pdu_meta->pdu->length;
+	/* A zero length PDU with LLID 0b01 (PDU_BIS_LLID_START_CONTINUE) would be a padding PDU.
+	 * However if there are errors in the PDU, it could be an incorrectly receive non-padding
+	 * PDU. Therefore only consider a PDU with erros as padding if received after the end
```
error
```suggestion
	 * PDU. Therefore only consider a PDU with error as padding if received after the end
```

niag-Oticon replied:
Done.

niag-Oticon committed:
bluetooth: controller: fixed problems in ISO-AL RX

[*] Fixed setting of SDU production status when exiting the
    ISOAL_ERR_SPOOL state for framed PDU consumption when receiving the
    start of the next SDU
[*] Changed condition identifying a padding PDU to include padding PDUs
    with errors received after the end fragment is seen and expanded
    unit tests to cover new condition
[*] Changed definition of ISOAL_PDU_STATUS_xxx errors to match value of
    similar errors defined for the SDU status

Signed-off-by: Nirosharn Amarasinghe <niag@demant.com>

niag-Oticon committed:
tests: bluetooth: controller: Add ISO AL unit test

Ztest based unit test of ISO adaptation layer.
Testing of unframed case, single SDU, recombined from
one and two PDUs.

Signed-off-by: Nirosharn Amarasinghe <niag@demant.com>

SebastianBoe reviewed (approved):
CMake changes are OK

mtpr-ot reviewed (approved):
LGTM
